# Cherry-pick from r488841.

--- a/base/time/time.h
+++ b/base/time/time.h
@@ -341,6 +341,14 @@ class TimeBase {
   // provided operators.
   int64_t ToInternalValue() const { return us_; }
 
+  // The amount of time since the origin (or "zero") point. This is a syntactic
+  // convenience to aid in code readability, mainly for debugging/testing use
+  // cases.
+  //
+  // Warning: While the Time subclass has a fixed origin point, the origin for
+  // the other subclasses can vary each time the application is restarted.
+  TimeDelta since_origin() const { return TimeDelta::FromMicroseconds(us_); }
+
   TimeClass& operator=(TimeClass other) {
     us_ = other.us_;
     return *(static_cast<TimeClass*>(this));
